airflow/www/views.py

```diff
@@ -1293,24 +1303,30 @@ def set_duration(tid):
     def graph(self, session=None):
         dag_id = request.args.get('dag_id')
         blur = conf.getboolean('webserver', 'demo_mode')
-        dag = dagbag.get_dag(dag_id)
-        if dag_id not in dagbag.dags:
-            flash('DAG "{0}" seems to be missing.'.format(dag_id), "error")
-            return redirect('/')
-
         root = request.args.get('root')
         if root:
-            dag = dag.sub_dag(
-                task_regex=root,
-                include_upstream=True,
-                include_downstream=False)
+            show_dag_id = root
+        else:
+            show_dag_id = dag_id
+        dag = session.query(models.DagModel).filter(models.DagModel.dag_id == show_dag_id).first()
```

Comment:
```suggestion
        dag = session.query(models.DagModel).filter(models.DagModel.dag_id == show_dag_id).one()
```

Reply:
Is this not raising an exception if not found? For that purpose I did use `.first()` and catch the None after it to give the user a nicer error

Reply:
But, this url should not be available to user user when the dag does not exist. So remove switch to `.one()` and remove the raise on the next line?

Reply:
Just testing... maybe you're in the middle of something, but simply running on this branch and opening the graph view of example_bash_operator gives me

```
......
  File "/airflow/airflow/www/views.py", line 1323, in graph
    .filter(models.DagRun.execution_date == dttm).one()
  File "/usr/local/lib/python3.7/site-packages/sqlalchemy/orm/query.py", line 3046, in one
    raise orm_exc.NoResultFound("No row was found for one()")
sqlalchemy.orm.exc.NoResultFound: No row was found for one()
```

Reply:
This means the user will get a 500 page

Reply:
Correct. So it should be fixed😛